File: 2023/day01/part1.ts
import { readFileSync } from "node:fs";

const lines = readFileSync("./input.txt", "utf-8")
	.replaceAll("\r", "")
	.split("\n");

const calibration_values = lines.map(line => {
	const digits = line.replaceAll(/\D/g, "");
	return digits.at(0)! + digits.at(-1)!;
}).map(Number);

console.log(calibration_values.reduce((a, b) => a + b));

File: 2025/day01/part1.py
with open("input.txt") as file:
	lines = [line.rstrip() for line in file]

count = 0
dial = 50
for instruction in lines:
	direction = instruction[0]
	number = int(instruction[1:])
	if direction == "L": number *= -1
	dial += number
	if dial % 100 == 0:
		count += 1
print(count)